app/controllers/application_controller.rb

```diff
@@ -135,9 +143,29 @@ def set_current_organization
   end
 
   def not_authorized
-    session[:user_return_to] = nil
-    flash[:notice] = "Sorry, you are not authorized to perform this action."
-    redirect_to(root_url)
+    message = "Sorry, you are not authorized to perform this action."
+    respond_to do |format|
+      format.json do
+        render json: {error: message}, status: :unauthorized
+      end
+      format.any do
+        session[:user_return_to] = nil
+        flash[:notice] = message
+        redirect_to(root_url)
+      end
+    end
+  end
+
+  def unsupported_media_type
+    respond_to do |format|
+      format.json do
+        render json: {error: "json unsupported"}, status: :unsupported_media_type
+      end
+      format.any do
+        flash[:alert] = "Page not found"
+        redirect_back_or_to root_url
+      end
+    end
```

Comment:
Above the `ActionController::UnknownFormat` for the new controllers, as well as any other unhandled format. Can add formats here (xml, csv, etc...) as needed.